stdlib/LinearAlgebra/test/diagonal.jl

```diff
@@ -234,6 +234,35 @@ Random.seed!(1)
         @test_throws ArgumentError triu(D, n + 2)
     end
 
+    @testset "testing istriu, istril, isdiag for Diagonal" begin
+        D1 = Diagonal([[1 0; 3 4], [5 0; 7 8]])
+        D2 = Diagonal([[1 2; 3 4], [5 6; 7 8]])
+        D3 = Diagonal([[1 2; 0 4], [5 6; 0 8]])
+        D4 = Diagonal([[1 0; 0 4], [5 0; 0 8]])
+        D5 = Diagonal([1, 2])
+
+        @test istril(D1) == true
+        @test istriu(D1) == false
+        @test isdiag(D1) == false
+
+        @test istril(D2) == false
+        @test istriu(D2) == false
+        @test isdiag(D2) == false
+
+        @test istril(D3) == false
+        @test istriu(D3) == true
+        @test isdiag(D3) == false
+
+        @test istril(D4) == true
+        @test istriu(D4) == true
+        @test isdiag(D4) == true
+
+        @test istril(D5) == true
+        @test istriu(D5) == true
+        @test isdiag(D5) == true
```

Comment:
@eulerkochy Tests seem to fail because you have overwritten previously defined `D1`, `D2` etc. Since you're testing block-diagonal matrices here, I'd suggest you remove the `testset-end` lines, and insert your tests including the `D1` etc. definitions (lines 238-262) after line 441, unless others find a better place. You could then skip the `D5` tests, because the "normal" case is already covered by lines 53, 56, and 57.

Reply:
Ok, cool!